### ISSUE TITLE
SQLCipher misleadingly prints the SQLite version

### ISSUE THREAD

**Issue by revolter** [sqlcipher/sqlcipher#354]
### Actual Behavior

`sqlcipher --version` looks like it should print the "version" of "sqlcipher", yet it prints the version of SQLite.

More so, `sqlcipher` prints:

```
SQLCipher version 3.30.1 2019-10-10 20:19:45
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
```

which clearly looks like "3.30.1" is the "version" of "SQLCipher".

All of these contradict with this comment: https://github.com/sqlcipher/sqlcipher/issues/353#issuecomment-584624369.

### Steps to Reproduce

SQLCipher version:

*Note:* If you are not posting a specific issue for the SQLCipher library, please consider posting your question to the SQLCipher [discuss site](https://discuss.zetetic.net/c/sqlcipher).  Thanks!

**Comment by sjlombardo** [comment]
Hello @revolter - The reason is that we leave the SQLite version number (e.g. 3.30.1) intact is that applications may use `sqlite3_libversion()`, `sqlite3_libversion_number()`, the `sqlite_version()`SQL Function, or the `SQLITE_VERSION`/`SQLITE_VERSION_NUMBER` macros. These are all ways that an application can identify at runtime or compile time what version of SQLite is in use. These version numbers may be used determine whether certain SQLite features are present that an application might rely on, or to establish a minimum version of SQLite that will work with an application. If we were to override those (i.e. return 4.x) then applications would either not be able to do such checks, or worse, when they did they may erroneously determine that the current version was above any such baseline (because SQLCipher version numbers are higher that SQLite's).

The recommended way to check the version of SQLCipher is to call [PRAGMA cipher_version](https://www.zetetic.net/sqlcipher/sqlcipher-api/#cipher_version). This will report the version of SQLCipher in use. If it is run against a non-SQLCipher library it will not return any result, making it equally useful for an application to ensure that SQLCipher is linked properly. This approach allows applications to determine both the SQLite and SQLCipher versions in a reliable, non-conflicting way. 

I acknowledge that this is not perfect, and it can be confusing when using the shell. However, given the impact, we do not plan to change the version number reporting.

**Comment by revolter** [comment]
What has anything you mentioned have to do with `sqlcipher --version`?

You mentioned C functions, I was talking about shell commands. Are there applications that rely on `sqlcipher --version` to identify at runtime or compile time what version of SQLite is in use?

Also, how does this stop you from changing the header of the `sqlcipher` shell command? I find it hard to believe that any app is using a REPL's output to detect the version.

**Comment by sjlombardo** [comment]
@revolter it has to do with it because the [shell command](https://sqlite.org/src/artifact/c2e20c43a44fb558) calls `sqlite3_libversion()` internally:

```
}else if( strcmp(z,"-version")==0 ){
      printf("%s %s\n", sqlite3_libversion(), sqlite3_sourceid());
      return 0;
}
```

**Comment by revolter** [comment]
But it could not call it internally. Or there could be a `--cipher-version` to mitigate the confusion.

What about the REPL?

**Comment by sjlombardo** [comment]
@revolter It depends on your definition of an application. It would not be unusual for a shell script to call the command line shell to determine whether a specific version is in available. We have no way to rule that out, and since SQLCipher has operated this way for years, it would be risky to change it. In addition, I don't see any situation where a user would be unable to get the information they need about either the SQLite or SQLCipher version. If there is a specific case that you are considering please elaborate with a specific example.

**Comment by revolter** [comment]
I see.

The problem is that this is objectively very confusing, and I just wanted to raise it from the perspective of a user with limited knowledge about the details or internals of this library. And I find it weird that you closed a valid issue, without at least waiting for my first reply...

Also, currently, you have no way of finding out the version of the installed SQLCipher CLI.

So I still think that at least the REPL could be changed to:
```diff
SQLCipher version 3.30.1 2019-10-10 20:19:45
+SQLCipher CLI version 4.3.0
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
sqlite>
```

and that there could be a separate `--cipher-version` argument that prints `4.3.0`.

**Comment by sjlombardo** [comment]
As mentioned in pull #357 the version information for SQLCipher is now included in a consistent way in v4.4.0 through shell header, the `--version` command line flag, or the `.version` shell command.

```
% ./sqlcipher --version
3.31.0 2020-01-22 18:38:59 f6affdd41608946fcfcea914ece149038a8b25a62bbe719ed2561c649b86alt1 (SQLCipher 4.4.0 community)
% ./sqlcipher
SQLite version 3.31.0 2020-01-22 18:38:59 (SQLCipher 4.4.0 community)
Enter ".help" for usage hints.
Connected to a transient in-memory database.
Use ".open FILENAME" to reopen on a persistent database.
sqlite> .version
SQLite 3.31.0 2020-01-22 18:38:59 f6affdd41608946fcfcea914ece149038a8b25a62bbe719ed2561c649b86alt1
SQLCipher 4.4.0 community
zlib version 1.2.11
clang-11.0.3
```
This approach ensures that both the SQLite and SQLCipher versions are discernible in a single line of output.
